engine/utils/collections/StackArray.h

```diff
@@ -0,0 +1,480 @@
+//
+//
+//  Copyright (c) 2022 Jonathan Moallem (@J-Mo63) & Aryeh Zinn (@Raelr)
+//
+//  This code is released under an unmodified zlib license.
+//  For conditions of distribution and use, please see:
+//      https://opensource.org/licenses/Zlib
+//
+
+#ifndef SIEGE_ENGINE_STACK_ARRAY_H
+#define SIEGE_ENGINE_STACK_ARRAY_H
+
+#include "ArrayUtils.h"
+#include "BitSet.h"
+#include "Iterators.h"
+
+namespace Siege::Utils
+{
+/**
+ * The Array (StackArray) is a simple wrapper around C arrays which adds some convenience (such
+ * as iterators and size tracking). The intention of this array is to 'replace' the standard C
+ * array within the codebase in places where it makes the most sense.
+ *
+ * This array class does NOT guarantee that data being retrieved is valid, and it is possible to
+ * retrieve garbage data. For a managed array type, please refer to the HeapArray or the MArray
+ * classes.
+ * @tparam T the type of the data being stored by the array.
+ * @tparam S the size of the array.
+ */
+template<typename T, size_t S>
+struct SArray
+{
+public:
+
+    /**
+     * A const overload to the subscript operator. Retrieves the element stored in the provided
+     * index.
+     * @param index the index to get the value from.
+     * @return the value stored in the provided index.
+     */
+    constexpr inline const T& operator[](const size_t& index) const
+    {
+        return Get(index);
+    }
+    /**
+     * An overload to the subscript operator. Retrieves the element stored in the provided index.
+     * @param index the index to get the value from.
+     * @return the value stored in the provided index.
+     */
+    constexpr inline T& operator[](const size_t& index)
+    {
+        return Get(index);
+    }
+
+    /**
+     * Checks if a given index is in bounds of the array.
+     * @param index the index to check.
+     * @return true if the index is in bounds, false if the index is not in bounds.
+     */
+    constexpr inline bool IsInBounds(const size_t& index)
+    {
+        return index < S;
+    }
+
+    /**
+     * Checks if a given index is in bounds of the array.
+     * @param index the index to check.
+     * @return true if the index is in bounds, false if the index is not in bounds.
+     */
+    constexpr inline const bool IsInBounds(const size_t& index) const
+    {
+        return index < S;
+    }
+
+    /**
+     * Retrieves the element stored in the provided index as a const reference.
+     * @param index the index to get the value from.
+     * @return the value stored in the provided index.
+     */
+    constexpr inline const T& Get(const size_t& index) const
```

Comment:
```suggestion
    constexpr inline const T& Get(size_t index) const
```
and so on

Reply:
Changed all instances of const size_t& .